assets/js/blocks/product-category/inspector-controls.tsx

```diff
@@ -0,0 +1,116 @@
+/**
+ * External dependencies
+ */
+import { InspectorControls } from '@wordpress/block-editor';
+import { PanelBody } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import ProductCategoryControl from '@woocommerce/editor-components/product-category-control';
+import GridLayoutControl from '@woocommerce/editor-components/grid-layout-control';
+import { getSetting } from '@woocommerce/settings';
+import GridContentControl from '@woocommerce/editor-components/grid-content-control';
+import ProductOrderbyControl from '@woocommerce/editor-components/product-orderby-control';
+import ProductStockControl from '@woocommerce/editor-components/product-stock-control';
+
+/**
+ * Internal dependencies
+ */
+import { Props } from './types';
+
+export interface InspectorControlsProps extends Props {
+	isEditing: boolean;
+	setChangedAttributes: ( changedAttributes: object ) => void;
+}
+
+export const ProductsByCategoryInspectorControls = (
+	props: InspectorControlsProps
+): JSX.Element => {
+	const { isEditing, attributes, setAttributes, setChangedAttributes } =
+		props;
+	const {
+		columns,
+		catOperator,
+		contentVisibility,
+		orderby,
+		rows,
+		alignButtons,
+		stockStatus,
+	} = attributes;
+
+	return (
+		<InspectorControls key="inspector">
+			<PanelBody
+				title={ __(
+					'Product Category',
+					'woo-gutenberg-products-block'
+				) }
+				initialOpen={ ! attributes.categories.length && ! isEditing }
+			>
+				<ProductCategoryControl
+					selected={ attributes.categories }
+					onChange={ ( value = [] ) => {
+						const ids = value.map( ( { id } ) => id );
+						const changes = { categories: ids };
+
+						// Changes in the sidebar save instantly and overwrite any unsaved changes.
+						setAttributes( changes );
+						setChangedAttributes( changes );
+					} }
+					operator={ catOperator }
+					onOperatorChange={ ( value = 'any' ) => {
+						const changes = { catOperator: value };
+						setAttributes( changes );
+						setChangedAttributes( changes );
+					} }
+					isCompact={ true }
+				/>
+			</PanelBody>
+			<PanelBody
+				title={ __( 'Layout', 'woo-gutenberg-products-block' ) }
+				initialOpen
+			>
+				<GridLayoutControl
+					columns={ columns }
+					rows={ rows }
+					alignButtons={ alignButtons }
```

Comment:
I guess that there is an error in the definition of the type of the props for the component `GridLayoutControl`. I think that we can replace [this](https://github.com/woocommerce/woocommerce-blocks/blob/5fc52a593e9ee2c3b6134f16e2e6b67ce2d9bcc0/assets/js/editor-components/grid-layout-control/index.js/#L16-L17) with `boolean`.